src/env.rs

```diff
@@ -115,6 +115,10 @@ pub(crate) fn fetch(opts: FetchOpts) -> Result<HashMap<String, String>, Error> {
         let user = prompt_default("Vault username: ", env::var("USER").ok())?;
         let password = prompt_password("Vault password: ")?;
         vault.ldap_auth(&user, &password)?;
+    } else if opts.kube && std::path::Path::new("/var/run/secrets/kubernetes.io/serviceaccount/token").exists() {
+        debug!("Authenticating with Vault via K8s ServiceAccount");
+        let role: String = service.to_string();
+        vault.kubernetes_auth(role)?;
```

Comment:
`service` is already a `String`, what you want here is to either give `kubernetes_auth` a copy of service, like `vault.kubernetes_auth(service.clone())?;`, or lend it a `&str` reference to `service` like `vault.kubernetes_auth(&service)?;`.

The second option requires a small change to `kubernetes_auth` (explained in later comments), but makes more sense in Rust, and avoids a needless copy.

In Rust when you give a function a value it is moved into that function, and you can't get it back, but using `&` lets you lend a value to a function in a way that you do get it back.